flocker/node/agents/ebs.py

```diff
@@ -195,11 +300,71 @@ def resize_volume(self, blockdevice_id, size):
         pass
 
     # cloud_instance_id here too
-    def attach_volume(self, blockdevice_id, host):
-        pass
+    def attach_volume(self, blockdevice_id, attach_to):
+        """
+        Attach an EBS volume to given compute instance.
+
+        :param unicode blockdevice_id: EBS UUID for volume to be attached.
+        :param unicode attach_to: Instance id of AWS Compute instance to
+            attached the blockdevice to.
+
+        :raises UnknownVolume: If there does not exist a BlockDeviceVolume
+            corresponding to the input blockdevice_id.
+        :raises AlreadyAttachedVolume: If the input volume is already attached
+            to a device.
+        """
+        volume = self._get(blockdevice_id)
+        if volume.attached_to is not None:
+            raise AlreadyAttachedVolume(blockdevice_id)
+
+        self.lock.acquire()
+        blockdevices = glob.glob('/sys/block/*')
```

Comment:
Something like `FilePath(b"/sys/block").children()` would be better here.  `*` is a simple enough glob that you don't really need globbing and `twisted.python.filepath.FilePath` should typically be preferred for path-related operations (instead of bare strings which are more error prone).